docs/rfc/001-Test-Environment-Infrastructure.md

```
@@ -0,0 +1,36 @@
+- Secret Management : Since secret management is bullet points of technical debts of previous attempt,related PR to this RFC is going to use cloud-init on secret management. See  [How security risks are prevented?](#to-provide-security--isolation-)
+- Documentation about usage & example cloud-init files must be provided.
+
+## To Provide Security & Isolation
+
+- Cloud-init must inject secrets as Environment vars and no secret shall be written to persistent files,
+- r/w access to Config files(e.g,CloudFlared) must be restricted,
+- Cloud-init must remove every evidence of secrets(e.g,caches,logs,histories),
+- SSH Root login must be disabled,Root execution privileges are restricted exclusively to the local CI runner via sudoers, with no external access permitted,
+- Generated image must run via QEMU for isolation.
+- Cloud-init must close all firewalls but loopback and CloudFlared & GitHub Actions Listener after run.Dependencies(e.g,DNS queries) must be allowed.
+  - This service must have `Before=network-pre.target` order.
```

markdownlint-cli2 (0.18.1)

Files should end with a single newline character

(MD047, single-trailing-newline)